### ISSUE TITLE
recode_road wrongly classifying Dual Carriage ways

### ISSUE THREAD

**Issue by hulsiejames** [udsleeds/openinfra#91]
recode_road function is sometimes wrongly classifying trunk roads that are not Dual Carriage ways as dual carriage way. 

The fix is to specify that a dual carriage way MUST have one_way="yes", whereas other important roads (such as the A66.) labeled highway="trunk" are important, but not dual carriage way as there is no one_way="yes" tag.

Just creating this issue as a note to self - I'll fix after meeting with Robin.

**Comment by hulsiejames** [comment]
[ab5f736](https://github.com/udsleeds/openinfra/commit/ab5f736a96e907afbcdc4e1961fef6a65393cdca)

^ Contains the propsed oi_recode_road_class function. The fucntion can now detect Dual Carriage ways (oneway key required) and A roads (Primary Roads)

Analysis is contained within the script, but to summarise: 
```
waldo::comapre(old, new)
>`old$Freq`: 2250 1474 566 1497 17176 430 2378
>`new$Freq`: 2250 1131 566 1840 17176 430 2378


# We can see that rows 2 & 4 have changed.
# Where 2 - Dual Carriage Ways and 
# 4 - Primary (A) Roads
# Thus, by changing our filtering we have moved 343 ways incorrectly classified 
# as dual carraige ways, back to being Primary Roads (this was due to A roads
# being marked as trunk or trunk_link) and I was not specifying that a Dual 
# Carriage way required a oneway key. 
```

**Comment by Robinlovelace** [comment]
Looks good, should be fixed soon!

**Comment by hulsiejames** [comment]
Fixed by [863a59c](https://github.com/udsleeds/openinfra/commit/863a59c52324a6736c6bfbf0130dc4c416a23fba)
